src/agent/testresultreader.ts

```diff
@@ -346,8 +303,42 @@ export class ResultReader {
                 durationInMs: "" + Math.round(testCaseDuration * 1000), //convert to milliseconds and round to nearest whole number since server can't handle decimals for test case duration
                 stackTrace: stackTrace
             };
+        }
+
+        if (runTitle && 0 !== runTitle.length) {
+            testSuiteSummary.name = this.fileNumberToTitle(runTitle, parseInt(fileNumber));
+        }
+        else if (rootNode.attributes().name) {
+            testSuiteSummary.name = rootNode.attributes().name;
+        }
+        else {
+            testSuiteSummary.name = this.runTitleJunit(fileName, parseInt(fileNumber));
+        }
+
+        if (rootNode.attributes().hostname) {
+            testSuiteSummary.host = rootNode.attributes().hostname;
+        }
 
-            testResults.push(testResult);
+        //assume runtimes from xml are current local time since timezone information is not in the xml. If xml date > current local date, fall back to local
+        if (rootNode.attributes().timestamp) {
+            var timestampFromXml = new Date(rootNode.attributes().timestamp);
+            if (timestampFromXml < new Date()) {
+                testSuiteSummary.timeStamp = timestampFromXml;
+            }
+        }
+
+        if (rootNode.attributes().time) {
+            totalRunDuration = parseFloat(rootNode.attributes().time); //in seconds
+        }
+        
+        //find test case nodes in JUnit result xml
+        var testResults = [];
+
+        if (rootNode.testcase) {
```

Comment:
This is a better edge case than the test I added, so I'll update the new test to reflect this situation. The right thing to do here would be to let the task pass and emit a warning message in the logs about there being no tests. I'm not sure what API emits log messages, but I'll dig into that a little bit.